src/convert.cpp

```diff
@@ -56,6 +56,12 @@ ddwaf_object* to_ddwaf_object_object(
   int depth,
   bool lim
 ) {
+  Napi::Value toJSON = obj.Get("toJSON");
+  if (toJSON.IsFunction()) {
+    Napi::Value json = toJSON.As<Napi::Function>().Call(obj, {});
+    obj = json.ToObject();
+  }
```

Comment:
maybe this should only be used in certain addresses only ? or if we should only check the ownProperty and not inherited props ? I'm a bit afraid of some prototype pollution here

Reply:
turns out JSON.stringify also uses the prototype so we're good